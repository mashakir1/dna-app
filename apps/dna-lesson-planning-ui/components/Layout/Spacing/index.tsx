// Styling
import styles from "./spacing.module.scss";

interface SpacingProps {
  height: "xs" | "s" | "m" | "l" | "xl" | "xxl";
}

const propToCssClassMap = {
  xs: styles.spacingXS,
  s: styles.spacingS,
  m: styles.spacingM,
  l: styles.spacingL,
  xl: styles.spacingXL,
  xxl: styles.spacingXXL
}

const Spacing: (props: SpacingProps) => JSX.Element = ({ height }) => {
  return <div className={propToCssClassMap[height]} />;
};

export default Spacing;
