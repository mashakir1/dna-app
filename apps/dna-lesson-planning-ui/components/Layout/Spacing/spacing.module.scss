.spacingXS {
  height: $spacing-xs;
}

.spacingS {
  height: $spacing-s;
}

.spacingM {
  height: $spacing-m;
}

.spacingL {
  height: $spacing-l;
}

.spacingXL {
  height: $spacing-xl;
}

.spacingXXL {
  height: $spacing-xxl;
}
