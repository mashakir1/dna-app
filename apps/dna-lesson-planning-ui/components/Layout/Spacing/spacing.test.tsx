import { render } from "@testing-library/react";
import Spacing from ".";

describe("Sidebar", () => {
  it("renders small height correctly", () => {
    const sidebar = render(<Spacing height={"s"} />).container;

    expect(sidebar).toMatchSnapshot()
  })

  it("renders large height correctly", () => {
    const sidebar = render(<Spacing height={"l"} />).container;

    expect(sidebar).toMatchSnapshot()
  })
})
