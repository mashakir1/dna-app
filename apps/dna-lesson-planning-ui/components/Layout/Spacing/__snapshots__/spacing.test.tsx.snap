// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders large height correctly 1`] = `
<div>
  <div
    class="spacingL"
  />
</div>
`;

exports[`Sidebar renders small height correctly 1`] = `
<div>
  <div
    class="spacingS"
  />
</div>
`;
